### ISSUE TITLE
Addition to the docs of the simple case of one graph, for node classification.

### ISSUE THREAD

**Issue by CodePanda9** [pyg-team/pytorch_geometric#1990]
This is related to [another question](https://github.com/rusty1s/pytorch_geometric/issues/1989) I posted, regarding node classification on a single graph, using your own custom dataset. However, I feel that this is a very fundamental topic, and it's missing from the documentation, preventing newcomers from getting a clear grasp (and utility) out of this beautiful library. 

In short, in the [create your own dataset](https://pytorch-geometric.readthedocs.io/en/latest/notes/create_dataset.html#frequently-asked-questions) part of the documentation, down below, an example is provided if a user does not want to use the `dataset` structure of pytorch. 

```
from torch_geometric.data import Data, DataLoader

data_list = [Data(...), ..., Data(...)]
loader = DataLoader(data_list, batch_size=32)
```
This works very well for graph classification. Just take your graphs, wrap each on in a `data` object, put all of them inside a python list, and you're good to go. It does not, however, work for the case of node level classification, when the user has a single graph (for example one created in networkx, or scraped off the web). A simple approach, such as:

```
# the user provides their own matrices
Data (x=x, edge_index=edge_index, edge_attr=edge_attr, y=y)   
# single data object
data_list = [Data(...)]   
# fixed train and test sizes (number of nodes), perform a split
train_dataset, test_dataset = torch.utils.data.random_split(data_list, [train_size, test_size]) 

train_loader = DataLoader(train_dataset,
                         batch_size=32,
                         shuffle=True)
test_loader = DataLoader(test_dataset,
                         batch_size=32,
                         shuffle=False)
```
will fail. 

I have still to find an alternative to this, and I believe it's too important to be left out of the documentation, hence the separate feature request. Sorry to insist upon this, I am before a deadline and I'm doing my best to make this work. 

Thank you once more.

**Comment by rusty1s** [comment]
For node classification on a single graph, you can find a [short tutorial](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html#learning-methods-on-graphs) in the documentation, which trains a model based on a subset of ground-truth node labels.

In addition, we provide multiple [Google Colabs](https://pytorch-geometric.readthedocs.io/en/latest/notes/colabs.html) for
* PyTorch Geometric Node Classification Introduction
* A Tutorial for Node Classification
* A Tutorial about Scalable Graph Neural Networks for Node Classification using Cluster-GCN.

In your case, I suggest to train your model without any sampling strategy similar to the GCN example in the Introduction. Since your model does not come with any pre-defined split, you need to create it manually:
```python
perm = torch.randperm(num_nodes]
train_idx = perm[:num_train_nodes]
test_idx = perm[num_train_nodes:]
```
With that, you can simply run your model like that:
```python
from torch_geometric.transforms import ToSparseTensor

data = ToSparseTensor()(data)  # Convert into a more memory-efficient format.

def train(data, train_idx):
    model.train()
    out = model(data.x, data.edge_index)
    loss = critertion(out[train_idx], data.y[train_idx]
```
If you encounter any memory issues, it might be a good idea to try the newly introduced [`SparseTensor` format](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html):
```python
from torch_geometric.transforms import ToSparseTensor

data = ToSparseTensor()(data)  # Convert into a more memory-efficient format.
model(data.x, data.adj_t)
```

**Comment by CodePanda9** [comment]
Thank you for this detailed answer. I will try it tomorrow, would be a good idea to post this simple later example in the docs I guess. High chance others like me will be stuck in the future :)

**Comment by CodePanda9** [comment]
> from torch_geometric.transforms import ToSparseTensor
> 
> data = ToSparseTensor()(data)  # Convert into a more memory-efficient format.
> out = model(data.x, data.adj_t)
> ```

I am dealing with a larger graph at the moment, that also has edge features (just 1 number per edge). How can I extract that from the adj_t matrix?

**Comment by rusty1s** [comment]
Edge features should be automatically added to `adj_t`. You can access them via:
```
row, col, edge_attr = adj_t.coo()
```
